ui/app/js/controllers/user_edit_controller.js

```diff
@@ -5,6 +5,88 @@ function ($scope, $modalInstance, CSRF, Permissions, users, is_edit, user) {
   $scope.loading = true;
   $scope.users = users;
   $scope.currentItemTab = 1;
+
+  var allPermissions = {
+        "admin": ["products"],
+        "release": ["actions", "products"],
+        "release_locale": ["actions", "products"],
+        "release_read_only": ["actions", "products"],
+        "rule": ["actions", "products"],
+        "permission": ["actions"],
+        "required_signoff": ["products"],
+        "scheduled_change": ["actions"],
+    };
+
+  $scope.onPermission = function(){
+      $scope.available_options = allPermissions[$scope.permission.permission];
+
+      // clear  all previous selections 
+      [
+          $scope.selected_options,
+          $scope.selected_actions,
+          $scope.selected_products
+      ] = [[], [], []];
+  };
+  $scope.selected_options = [];
+
+  $scope.actions = ["create", "modify", "enact"];
+  $scope.selected_actions = [];
+  $scope.action = null;
+
+  $scope.products = [];
+  $scope.selected_products = [];
+  $scope.product = null;
+
+  // get products in releases
+  Permissions.getProducts().success(function(response) {
+    $scope.products = response.product;
+  });
+
+  $scope.$watch(function(){
```

Comment:
I don't think this is how $scope.$watch is supposed to work...the first argument to it is a watchExpression, which is thing that determines whether or not to run some code when something changes, _not_ something that makes changes. You can see an example in Balrog's codebase here: https://github.com/mozilla/balrog/blob/master/ui/app/js/controllers/rules_controller.js#L115

It's possible you may want multiple $watch expressions to handle options, products, and actions.